src/types/api.js

```diff
@@ -17,6 +17,32 @@
  * @prop {string} text
  * @prop {string} uri
  * @prop {string} user
+ * @prop {boolean} hidden
+ * @prop {boolean} [$highlight]
+ * @prop {boolean} [$orphan]
+ * @prop {boolean} [$anchorTimeout]
+ *
+ * @prop {Object} document
+ *   @prop {string} document.title
+ *
+ * @prop {Object} permissions
+ *   @prop {string[]} permissions.read
+ *   @prop {string[]} permissions.update
+ *   @prop {string[]} permissions.delete
+ *
+ * @prop {Object[]} target
+ *   @prop {string} target.source
+ *   @prop {Object[]} target.selector
+ *     @prop {string} target.selector.type
+ *     @prop {string} target.selector.exact
+ *     @prop {number} target.selector.start
+ *
+ * @prop {Object} moderation
```

Comment:
```suggestion
 * @prop {Object} [moderation]
```

The `moderation` parameter may be missing.

Reply:
Related to this, I expected that if a property is marked as optional then you'd get an error if you try to use it without checking whether it exists or not. That doesn't appear to be the case though. I'm unclear if that is [intended behaviour or a bug](https://github.com/microsoft/TypeScript/issues/39111), as this is different to how it works in `.ts` files.

In any case, it is useful for human readers to convey whether a property is optional or not.